### ISSUE TITLE
Unfortunately, (AppsName) has stopped.

### ISSUE THREAD

**Issue by EdoEscada** [DrKLO/Telegram#1359]
Dear friends, I have produced the APK and then run it in my Android smartphone. The pop up error is : Unfortunately, (AppsName) has stopped, happened in my 3 Android Smartphones.

Following some internet advises, I have cleared the app cache, clear RAM, restarted my mobile, set my Google Play Store not to update APK, But the problem still showing.l 

I use Android Studio the latest version.
Hoping your advise so crazy:)

Thx
Regards

**Comment by EdoEscada** [comment]
Okay got some points.
I have changed the values in BuildVars.Java

Now regarding my API Data Center given by the system, like this:

Available MTProto servers
Test configuration:149.154.167.40:443
Production configuration:149.154.167.50:443

I am opening ConnectionManager.Java and found:

    private void fillDatacenters() {
        if (datacenters.size() == 0) {
            if (isTestBackend == 0) {
                Datacenter datacenter = new Datacenter();
                datacenter.datacenterId = 1;
                datacenter.addAddressAndPort("149.154.175.50", 443, 0);
                datacenter.addAddressAndPort("2001:b28:f23d:f001:0000:0000:0000:000a", 443, 1);
                datacenters.put(datacenter.datacenterId, datacenter);

                datacenter = new Datacenter();
                datacenter.datacenterId = 2;
                datacenter.addAddressAndPort("149.154.167.51", 443, 0);
                datacenter.addAddressAndPort("2001:67c:4e8:f002:0000:0000:0000:000a", 443, 1);
                datacenters.put(datacenter.datacenterId, datacenter);

                datacenter = new Datacenter();
                datacenter.datacenterId = 3;
                datacenter.addAddressAndPort("149.154.175.100", 443, 0);
                datacenter.addAddressAndPort("2001:b28:f23d:f003:0000:0000:0000:000a", 443, 1);
                datacenters.put(datacenter.datacenterId, datacenter);

                datacenter = new Datacenter();
                datacenter.datacenterId = 4;
                datacenter.addAddressAndPort("149.154.167.91", 443, 0);
                datacenter.addAddressAndPort("2001:67c:4e8:f004:0000:0000:0000:000a", 443, 1);
                datacenters.put(datacenter.datacenterId, datacenter);

                datacenter = new Datacenter();
                datacenter.datacenterId = 5;
                datacenter.addAddressAndPort("149.154.171.5", 443, 0);
                datacenter.addAddressAndPort("2001:b28:f23f:f005:0000:0000:0000:000a", 443, 1);
                datacenters.put(datacenter.datacenterId, datacenter);
            } else {
                Datacenter datacenter = new Datacenter();
                datacenter.datacenterId = 1;
                datacenter.addAddressAndPort("149.154.175.10", 443, 0);
                datacenter.addAddressAndPort("2001:b28:f23d:f001:0000:0000:0000:000e", 443, 1);
                datacenters.put(datacenter.datacenterId, datacenter);

                datacenter = new Datacenter();
                datacenter.datacenterId = 2;
                datacenter.addAddressAndPort("149.154.167.40", 443, 0);
                datacenter.addAddressAndPort("2001:67c:4e8:f002:0000:0000:0000:000e", 443, 1);
                datacenters.put(datacenter.datacenterId, datacenter);

                datacenter = new Datacenter();
                datacenter.datacenterId = 3;
                datacenter.addAddressAndPort("149.154.175.117", 443, 0);
                datacenter.addAddressAndPort("2001:b28:f23d:f003:0000:0000:0000:000e", 443, 1);
                datacenters.put(datacenter.datacenterId, datacenter);
            }
        } else if (datacenters.size() == 1) {
            Datacenter datacenter = new Datacenter();
            datacenter.datacenterId = 2;
            datacenter.addAddressAndPort("149.154.167.51", 443, 0);
            datacenter.addAddressAndPort("2001:67c:4e8:f002:0000:0000:0000:000a", 443, 1);
            datacenters.put(datacenter.datacenterId, datacenter);

            datacenter = new Datacenter();
            datacenter.datacenterId = 3;
            datacenter.addAddressAndPort("149.154.175.100", 443, 0);
            datacenter.addAddressAndPort("2001:b28:f23d:f003:0000:0000:0000:000a", 443, 1);
            datacenters.put(datacenter.datacenterId, datacenter);

            datacenter = new Datacenter();
            datacenter.datacenterId = 4;
            datacenter.addAddressAndPort("149.154.167.91", 443, 0);
            datacenter.addAddressAndPort("2001:67c:4e8:f004:0000:0000:0000:000a", 443, 1);
            datacenters.put(datacenter.datacenterId, datacenter);

            datacenter = new Datacenter();
            datacenter.datacenterId = 5;
            datacenter.addAddressAndPort("149.154.171.5", 443, 0);
            datacenter.addAddressAndPort("2001:b28:f23f:f005:0000:0000:0000:000a", 443, 1);
            datacenters.put(datacenter.datacenterId, datacenter);
        }
    }


WHERE do I have to put my : 

Available MTProto servers
Test configuration:149.154.167.40:443
Production configuration:149.154.167.50:443

????????????????

Please help.

Thank you

**Comment by DrKLO** [comment]
You don't need to change anything in network code. By the way, this code is already outdated.

**Comment by EdoEscada** [comment]
Yes, Doc, I just downloaded the latest code few hours ago, Thank you

**Comment by EdoEscada** [comment]
Now I have the latest version of source.
Using Android Studio.

I changed BuildVars to be related with my ID like this:

public class BuildVars {
    public static boolean DEBUG_VERSION = false;
    public static int BUILD_VERSION = 627;
    public static int APP_ID = 41367; //obtain your own APP_ID at https://core.telegram.org/api/obtaining_api_id
    public static String APP_HASH = "b887f8aa3e1b0a7e7a67468eadab50b2"; //obtain your own APP_HASH at https://core.telegram.org/api/obtaining_api_id
    public static String HOCKEY_APP_HASH = "your-hockeyapp-api-key-here";
    public static String GCM_SENDER_ID = "760348033672";
    public static String SEND_LOGS_EMAIL = "edo.escada@gmail.com";
    public static String BING_SEARCH_KEY = ""; //obtain your own KEY at https://www.bing.com/dev/en-us/dev-center
    public static String FOURSQUARE_API_KEY = ""; //obtain your own KEY at https://developer.foursquare.com/
    public static String FOURSQUARE_API_ID = ""; //obtain your own API_ID at https://developer.foursquare.com/
    public static String FOURSQUARE_API_VERSION = "20150326";

Then compiled it to be an APK, as is. Success.
But still stopped by my KitKat :(. "Unfortunately, apss has stopped"
I really dont know that is stopped by server or by my KitKat.
I dont have GCM Id yet, does that influence to the process ?
If I put GCM Id with blank or "" .. what will be happened.. wrong ?
Now: "public static String GCM_SENDER_ID = "760348033672";"
I did not change anymore... Just compiled it as is.
Is there possibility conflict with any ID with main original Telegram Apk or something like that ?
Would you like pls to advise a little more, @DrKLO .. Thank you
